src/cluster_legacy.c

```diff
@@ -2042,6 +2042,23 @@ static void getClientPortFromGossip(clusterMsgDataGossip *g, int *tls_port, int
     }
 }
 
+/* Returns a string with the byte representation of the node ID (i.e. nodename)
+ * along with 8 trailing bytes for debugging purposes. */
+char *getCorruptedNodeIdByteString(clusterMsgDataGossip *gossip_msg) {
+    const int num_bytes = CLUSTER_NAMELEN + 8;
+    /* Allocate enough room for 4 chars per byte + null terminator */
+    char *byte_string = (char*) zmalloc((num_bytes*4) + 1); 
```

Comment:
Do we need a dynamic allocation for this ?

Reply:
I preferred it to have a simpler API since it was an exceptional case since I didn't want the caller to have knowledge about the max size of the response.